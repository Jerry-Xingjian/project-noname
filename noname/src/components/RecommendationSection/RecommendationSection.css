.recommendation-my-info .avatar-image {
  width: 64px;
  height: 64px;
}

.suggestion-text {
  font-weight: lighter;
  font-size: 1rem;
  text-align: end;
  margin: 0;
}

.see-all-button {
  text-align: end !important;
  padding: 0 !important;
}