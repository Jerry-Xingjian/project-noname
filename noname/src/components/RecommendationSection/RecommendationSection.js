/* eslint-disable no-underscore-dangle */
import React, { useState, useEffect } from 'react';
import RecommendationCard from '../RecommendationCard/RecommendationCard';
import './RecommendationSection.css';
import { getUserRecommendations } from '../../utils/api/user';

function RecommendationSection(props) {
  const [recommendations, setRecommendations] = useState([]);
  const { userInfo } = props;

  useEffect(() => {
    async function fetchRecommendations() {
      // console.log('trigger fetch recommendation');
      // eslint-disable-next-line no-underscore-dangle
      const rawData = await getUserRecommendations(userInfo._id);
      setRecommendations(rawData.data.data.remdUsers);
    }
    if ('username' in userInfo) {
      fetchRecommendations();
    }
  }, [userInfo]);

  return (
    <div className="container d-flex justify-content-between align-items-center mt-2 mb-4">
      <div className="recommendation-left d-flex flex-column justify-content-end">
        <div className="recommendation-my-info d-flex align-middle align-items-center justify-content-end">
          <img
            className="avatar-image ms-3 my-2 mx-2"
            src={userInfo.profilePicture}
            alt="Avatar"
            mode="aspectFit"
          />
          <h4 className="card-title ms-4">{ userInfo.username }</h4>
        </div>
        <div className="d-flex flex-column mt-2">
          <p className="suggestion-text">
            <strong>Top Two </strong>
            suggestions
            <br />
            for you
            <strong> TODAY</strong>
          </p>
          {/* <button type="button" className="btn btn-link text-decoration-none
          py-0 see-all-button">See All</button> */}
        </div>
      </div>
      <div className="d-flex flex-column mt-3">
        {
          recommendations.length === 0 ? (
            <p className="text-muted">No Recommentdations for you yet</p>
          ) : (
            recommendations.slice(0, 2).map((recommendation) => (
              <RecommendationCard
                key={recommendation._id}
                recommendedInfo={recommendation}
                userInfo={userInfo}
              />
            ))
          )
        }
      </div>
    </div>
  );
}

export default RecommendationSection;
