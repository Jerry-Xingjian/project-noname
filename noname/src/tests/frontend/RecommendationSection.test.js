import { render, screen } from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom'
import RecommendationSection from '../../components/RecommendationSection/RecommendationSection';
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer';

const userInfo = {
	"_id": "6388320305d016d46329331d",
	"email": "qhao@seas.upenn.edu",
	"password": "$2b$10$Mxx3KERe2hI9GUEkb8Hsm.jr2gLNJyZYfooiGYK1RW0QBPdComYSO",
	"username": "qhao",
	"profilePicture": "https://nonome-project-media.s3.amazonaws.com/default_avatar.png",
	"followings": [],
	"followers": [],
	"createdAt": "2022-12-01T04:48:03.298Z",
	"updateAt": "2022-12-01T04:48:03.298Z",
	"bio": "This is some bio of the user qhao.\nThis can be super long but it's fine."
}

describe("snapshot test", () => {
  test('RecommendationCard matches snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
          <RecommendationSection userInfo={userInfo} />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe("Element existence test", () => {
  test('renders Follow text', () => {
    render(
      <BrowserRouter>
          <RecommendationSection userInfo={userInfo} />
      </BrowserRouter>
    );
    const text = screen.getByText('Top Two');
    expect(text).toBeInTheDocument();
    expect(text.innerHTML).toBe('Top Two ');
  });
})