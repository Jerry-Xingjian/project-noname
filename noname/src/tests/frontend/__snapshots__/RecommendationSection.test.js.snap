// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test RecommendationCard matches snapshot 1`] = `
<div
  className="container d-flex justify-content-between align-items-center mt-2 mb-4"
>
  <div
    className="recommendation-left d-flex flex-column justify-content-end"
  >
    <div
      className="recommendation-my-info d-flex align-middle align-items-center justify-content-end"
    >
      <img
        alt="Avatar"
        className="avatar-image ms-3 my-2 mx-2"
        mode="aspectFit"
        src="https://nonome-project-media.s3.amazonaws.com/default_avatar.png"
      />
      <h4
        className="card-title ms-4"
      >
        qhao
      </h4>
    </div>
    <div
      className="d-flex flex-column mt-2"
    >
      <p
        className="suggestion-text"
      >
        <strong>
          Top Two 
        </strong>
        suggestions
        <br />
        for you
        <strong>
           TODAY
        </strong>
      </p>
    </div>
  </div>
  <div
    className="d-flex flex-column mt-3"
  />
</div>
`;
